ICSharpCode.Decompiler/CSharp/CallBuilder.cs

```diff
@@ -109,6 +109,15 @@ public TranslatedExpression Build(CallInstruction inst)
 					expectedTargetDetails.NeedsBoxingConversion = true;
 				}
 			}
+			
+			// Special case to deal with implicit dynamic conversions:
+			bool allowDynamicDispatch = true;
```

Comment:
When does it ever make sense to `allowDynamicDispatch` for a static `CallInstruction`?
I'd say that dynamic dispatch should never be allowed (which in turn means we can convert `dynamic` arguments unconditionally before we even try OR)